docs/x509.rst

```diff
@@ -196,6 +196,24 @@ X.509 Certificate Object
             >>> isinstance(cert.signature_hash_algorithm, hashes.SHA256)
             True
 
+    .. attribute:: extensions
+
+        :type: :class:`collections.OrderedDict`
```

Comment:
I think I'd prefer to expose our own class, since that better hides
implementation details.

On Sat, Mar 21, 2015 at 8:50 PM, Paul Kehrer <notifications@github.com>
wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1769#discussion_r26898645>:
>
> > @@ -196,6 +196,24 @@ X.509 Certificate Object
> >              >>> isinstance(cert.signature_hash_algorithm, hashes.SHA256)
> >              True
> >
> > +    .. attribute:: extensions
> > +
> > +        :type: :class:`collections.OrderedDict`
>
> We discussed that a bit in this PR. OrderedDict captures the behavior we
> want here very well (preserve order, but at most one of each), so I'd like
> to use that, but I'd be okay with creating an Extensions that has a
> get_by_oid to match what we do on Name (but also implements __get_item__).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1769/files#r26898645>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084